scripts/batchProcessRasters.R

```diff
@@ -144,102 +149,273 @@ batchImportRasters_toOSW<-function(inpJSON){
+	}
+	
+	#create log file for this run:
+	filelogn<-paste0("OSW_CA_batchRasterProcessing_",logstart)
+	logfile<-paste(localTempDir,filelogn,".log",sep="")
+	savelog<-paste0(logDir,filelogn,".log")
+	sitd<-as.character(sessionInfo())
+	
+	zz <- try(file(logfile, "w"),silent=T)
+	flog<-0
+	if(inherits(zz,"try-error")){
+		retJSON<-"{\"Out\":[{\"Result\":\"Error\",\"Process\":\"Open local log file\",\"value\":\"Failed to create log file in local temp directory\"}], \"Description\":\"Could not open log file in local directory. Logs will not be kept.\"}"
+		flog<-1
+	}
+	if(flog==0){
+		cat("Log batch raster processing script", paste("Started", logstart), file = zz, sep = "\n", append=TRUE)
+		cat("\n","\n",file = zz, append=TRUE)
+		cat("All libraries required are loaded.",sep ="\n",file = zz, append=TRUE)
+		cat("Valid JSON string provided.",sep ="\n",file = zz, append=TRUE)
+		cat(paste("Processing YAML file:",inpYAML),sep ="\n",file = zz, append=TRUE)
+		cat("YAML contains valid and sufficient input.",sep ="\n",file = zz, append=TRUE)
+	}
+	
+	# copy raster data locally from bucket
+	wgrd<-try(save_object(object="FirstProcess/AliquoteGrid/aliqgrid.grd",bucket="offshore-wind-data",file=paste0(localTempDir,"aliqgrid.grd")),silent=T)
+	wgri<-try(save_object(object="FirstProcess/AliquoteGrid/aliqgrid.gri",bucket="offshore-wind-data",file=paste0(localTempDir,"aliqgrid.gri")),silent=T)
+	if(inherits(wgrd,"try-error") | inherits(wgri,"try-error"))
+	if(!inherits(wgrd,"try-error") & !inherits(wgri,"try-error")){
+		retJSON<-"{\"Out\":[{\"Result\":\"Error\",\"Process\":\"Copy base grid into temp directory\",\"value\":\"Failed\"}], \"Description\":\"Failed to copy base grid into the local temp directory\"}"
+		if(flog==0){
+			cat("Failed to copy base grid locally.",sep ="\n",file = zz, append=TRUE)
+			if(logSessionInfo){
+				cat(sitd,sep="\n",file=zz,append=TRUE)
+			}
+			cat("\n","\n",file = zz, append=TRUE)
+			cat("Terminating batch raster processing.",sep ="\n",file = zz, append=TRUE)
+			close(zz)
+			svlg<-try(put_object(file=logfile,object=savelog,bucket="offshore-wind-data"),silent=TRUE)
+		}
+		return(retJSON)
+	}
+	fegrd<-file.exists(paste0(localTempDir,"aliqgrid.grd"))
+	fegri<-file.exists(paste0(localTempDir,"aliqgrid.grd"))
+	if(!fegrd | !fegri){
+		retJSON<-"{\"Out\":[{\"Result\":\"Error\",\"Process\":\"Copy base grid into temp directory\",\"value\":\"Failed\"}], \"Description\":\"Failed to copy base grid into the local temp directory\"}"
+		if(flog==0){
+			cat("Failed to copy base grid locally.",sep ="\n",file = zz, append=TRUE)
+			if(logSessionInfo){
+				cat(sitd,sep="\n",file=zz,append=TRUE)
+			}
+			cat("\n","\n",file = zz, append=TRUE)
+			cat("Terminating batch raster processing.",sep ="\n",file = zz, append=TRUE)
+			close(zz)
+			svlg<-try(put_object(file=logfile,object=savelog,bucket="offshore-wind-data"),silent=TRUE)
+		}
+		return(retJSON)
+	}
+
+	aliqgrd<-try(raster(paste0(localTempDir,"aliqgrid.gri")),silent=T)

```

Comment:
`aliqgrid.gri` and `aliqgrid.grd` are used quite a few times in this code. This is a good case to use a global variable in place of a string literal. It will eliminate any spelling errors and if you pick good names, will make it clear as to which file you're using.

Reply:
I'll not doing this. Maybe I am not understanding what you said, but let's be clear that they are variables, not string literals, and represent objects that are not strings. Not sure if R looks first in the global environment or the local/current environment. I think it goes from local to encompassing, to encompassing,... all the way to global. As to the names, I like aliqgrid.